src/main/java/org/mastodon/mamut/feature/branch/sinuosity/BranchSinuosityFeatureComputer.java

```diff
@@ -0,0 +1,105 @@
+package org.mastodon.mamut.feature.branch.sinuosity;
+
+import org.mastodon.graph.algorithm.RootFinder;
+import org.mastodon.graph.algorithm.traversal.DepthFirstIterator;
+import org.mastodon.mamut.feature.MamutFeatureComputer;
+import org.mastodon.mamut.model.Spot;
+import org.mastodon.mamut.model.branch.BranchLink;
+import org.mastodon.mamut.model.branch.BranchSpot;
+import org.mastodon.mamut.model.branch.ModelBranchGraph;
+import org.mastodon.mamut.util.GeometryUtils;
+import org.mastodon.properties.DoublePropertyMap;
+import org.scijava.ItemIO;
+import org.scijava.plugin.Parameter;
+import org.scijava.plugin.Plugin;
+
+import java.util.Arrays;
+import java.util.Iterator;
+
+/**
+ * Computes {@link BranchSinuosityFeature}
+ */
+@Plugin( type = MamutFeatureComputer.class )
+public class BranchSinuosityFeatureComputer implements MamutFeatureComputer
+{
+
+	@Parameter
+	private ModelBranchGraph branchGraph;
+
+	@Parameter( type = ItemIO.OUTPUT )
+	private BranchSinuosityFeature output;
+
+	@Override
+	public void createOutput()
+	{
+		if ( null == output )
+			output = new BranchSinuosityFeature( new DoublePropertyMap<>( branchGraph.vertices().getRefPool(), 0 ) );
+	}
+
+	@Override
+	public void run()
+	{
+		computeBranchSinuosity();
+	}
+
+	private void computeBranchSinuosity()
+	{
+		DepthFirstIterator< BranchSpot, BranchLink > depthFirstIterator = new DepthFirstIterator<>( branchGraph );
+		for ( BranchSpot root : RootFinder.getRoots( branchGraph ) )
```

Comment:
Thanks for pointing this out.
It was needed, since I wanted to compute the sinuosity including the state before the cell division, which I decided now not to do.

Reply:
Why was depth first iteration needed? I don't get it. Finding if a branch spot is root can be done with of depth first iteration. And finding the coordinates of the previous spot can also be done without depth first iteration.